common/crypto_utils/avalon_crypto_utils/signature.py

```diff
@@ -269,21 +269,18 @@ def generate_client_signature(
             input_json_params, session_key,
             session_iv, worker.encryption_key, data_key, data_iv)
 
-        if input_json_params["requesterNonce"] and \
-                is_valid_hex_str(input_json_params["requesterNonce"]):
-            nonce = crypto.string_to_byte_array(
-                input_json_params["requesterNonce"])
-        else:
+        if "requesterNonce" in input_json_params and \
+                len(input_json_params["requesterNonce"]) == 0:
```

Comment:
If user passes "requesterNonce" then shouldn't we add a check for valid hex string ?